tests/test-dladdr.c

```diff
@@ -0,0 +1,157 @@
+#include <stdio.h>
+#include <string.h>
+
+/* required for non Windows builds */
+#define __USE_GNU
```

Comment:
Linux documentation for dlopen says that you should `#define _GNU_SOURCE` and not `__USE_GNU`. So what is the reason for __USE_GNU?

Reply:
On my openSUSE distribution I found 
```
cat /usr/include/dlfcn.h
...
#ifdef __USE_GNU
/* Structure containing information about object searched using
   `dladdr'.  */
typedef struct
{
  const char *dli_fname;        /* File name of defining object.  */
  void *dli_fbase;              /* Load address of that object.  */
  const char *dli_sname;        /* Name of nearest symbol.  */
  void *dli_saddr;              /* Exact value of nearest symbol.  */
} Dl_info;

/* Fill in *INFO with the following information about ADDRESS.
   Returns 0 iff no shared object's segments contain that address.  */
extern int dladdr (const void *__address, Dl_info *__info)
     __THROW __nonnull ((2));
...
#endif

Reply:
`__USE_GNU` seems like internal GNU C macro. Really `_GNU_SOURCE` should be used.

Reply:
I can make the comment in line 4 more specific.

Reply:
I found the reason:  _GNU_SOURCE must be defined in front of the first system include, otherwise the prototype for dladdr() will not be present.

Reply:
Yes, this is correct. See `man feature_test_macros` for more details.